vpr/src/device/rr_graph_view.h

```diff
@@ -62,6 +62,16 @@ class RRGraphView {
         return node_storage_.node_capacity(node);
     }
 
+    /* Get the direction of a routing resource node. This function is inlined for runtime optimization. */
+    inline e_direction node_direction(RRNodeId node) const {
+        return node_storage_.node_direction(node);
+    }
+
+    /* Get the direction string of a routing resource node. This function is inlined for runtime optimization. */
+    inline const char* node_direction_string(RRNodeId node) const {
```

Comment:
Hi @vaughnbetz @hzeller  I do not have a strong opinion here. Personally, I prefer the C++ string object as it is easier to manipulate (e.g., check empty, check value etc.). For internal storage, we can use the ``const char*`` to save memory footprint.  For data exchange, using the string object is fine because it does not cost too much.

Reply:
I prefer having the method over having a constant array of strings that we expose (better abstraction). 

 I also have a mild preference for the C++ string object due to easier manipulation; so far that means Xifan and I both have a mild preference to change this method to return a C++ string.

Reply:
I have altered the code to use a C++ std::string. Let me know if you see anything that still needs to be adjusted. Thanks!

Reply:
So if you want to do a string return, I'd not do it this way as the compiler will have to copy every time (it does return value optimization, but that only means that it copies it in the string on the call-site).

If these methods are only called every now and then for printing something, this is ok. If the strings are very short (I think they are) it probably is also not too big of a deal (as the std::string does not allocate dynamic memory in that case).

Something to consider: if these strings can be pre-allocated as `std:string something[arraysize]`, you can return `const std::string&` references and don't have to worry about copy.

In which context are the returned strings needed ? Are you doing string manipulation on these that require std::string-ness ? Would you have to call `.c_str()` on every string on reception, because it is used in some fprintf()-like context ?